vm/systemSmartContracts/eei.go

```diff
@@ -433,8 +433,8 @@ func (host *vmContext) transferBeforeInternalExec(callInput *vmcommon.ContractCa
 		logEntry := &vmcommon.LogEntry{
 			Identifier: []byte(transferValueOnly),
 			Address:    callInput.RecipientAddr,
-			Topics:     [][]byte{sender, callInput.RecipientAddr, callInput.CallValue.Bytes()},
-			Data:       []byte{},
+			Topics:     [][]byte{sender, callInput.CallValue.Bytes()},
+			Data:       vmcommon.FormatLogDataForCall("", callInput.Function, callInput.Arguments),
```

Comment:
we need full integration here as well.  You can delete the callInput.CallValue.Cmp(zero).
This function is called from 2 places - either ExecuteOnDestContext or DeploySystemSC - add those endpoint names as the first element of data.